File: my-drive-client/src/Core/Folder.ts
export type Folder = {
    id: string,
    name: string,
    path: string,
    files: File[],
    subFolders: Folder[],
    isAccessible: boolean,
    createdAt: Date,
    size: number,
    parentFolderId: string | null,
};

export type CreateFolderDto = {
    id: string,
    name: string,
    path: string,
    isAccessible: boolean,
    parentFolderId: string
};

export type EditFolderDto = {
    id: string,
    name: string,
    path: string,
    isAccessible: boolean,
    parentFolderId: string
}

export type UploadFolderDto = {
    files: File[],
    isAccessible: boolean,
    parentFolderId: string
}

File: Models/Folder.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace MyDrive.Models;

public class Folder
{
    [Key]
    public Guid Id { get; set; }

    [Required]
    [StringLength(256, MinimumLength = 1)]
    public required string Name { get; set; }

    [Required]
    [StringLength(512, MinimumLength = 1)]
    public required string Path { get; set; }

    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Required]
    public bool IsAccessible { get; set; } = false;
    
    public long Size { get; set; }
    public Guid? ParentFolderId { get; set; }
    [JsonIgnore]
    public Folder? ParentFolder { get; set; }
    public List<Folder> SubFolders { get; set; } = new();
    public List<FileType> Files { get; set; } = new();
}